### PR TITLE
feat(registers) add registers search

### PR DESCRIPTION
**Pull request by roy1910**
https://tillhub.atlassian.net/browse/DAS-1038

### CODE REVIEW

File: src/v1/registers.ts [hunk 1]
```diff
@@ -158,4 +164,28 @@ export class Registers extends ThBaseHandler {
...
+
+  async search (query: string | SearchQuery): Promise<RegisterResponse> {
+    let uri
+    if (typeof query === 'string') {
+      uri = this.uriHelper.generateBaseUri(`/search?q=${query}`)
+    } else if (typeOf(query) === 'object') {
+      const base = this.uriHelper.generateBaseUri('/search')
+      uri = this.uriHelper.generateUriWithQuery(base, query)
+    } else {
+      throw new errors.RegistersSearchFailed('Could not search for branch - query type is invalid')
```
Comment (erezsob) [inline src/v1/registers.ts:176]:
```suggestion
      throw new errors.RegistersSearchFailed('Could not search for register - query type is invalid')
```

**Review by erezsob — APPROVED** [review]
Great!

Please remember to do a regular merge (NOT squash).

File: src/v1/registers.ts (hunk 1 above)
Comment (roy1910) [reply src/v1/registers.ts]: fixed

**Pull request merged**